driver_test.go

```diff
@@ -338,3 +309,356 @@ func TestIsDdl(t *testing.T) {
 		}
 	}
 }
+
+func TestExecContextDml(t *testing.T) {
+
+	// Open db.
+	ctx := context.Background()
+	db, err := sql.Open("spanner", dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	// Set up test table.
+	_, err = db.ExecContext(ctx,
+		`CREATE TABLE TestExecContextDml (
+			key	INT64,
+			testString	STRING(1024),
+			testBytes	BYTES(1024),
+			testInt	INT64,
+			testFloat	FLOAT64,
+			testBool	BOOL
+		) PRIMARY KEY (key)`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type testExecContextDmlRow struct {
+		key        int
+		testString string
+		testBytes  []byte
+		testInt    int
+		testFloat  float64
+		testBool   bool
+	}
+
+	type then struct {
+		query     string
+		wantRows  []testExecContextDmlRow
+		wantError bool
+	}
+
+	tests := []struct {
+		name  string
+		given string
+		when  string
+		then  then
+	}{
+		{
+			name: "insert single tuple",
+			when: `
+				INSERT INTO TestExecContextDml 
+				(key, testString, testBytes, testInt, testFloat, testBool)
+				VALUES (1, "one", CAST("one" as bytes), 42, 42, true ) `,
+			then: then{
+				query: `SELECT * FROM TestExecContextDml WHERE key = 1`,
+				wantRows: []testExecContextDmlRow{
+					{key: 1, testString: "one", testBytes: []byte("one"), testInt: 42, testFloat: 42, testBool: true},
+				},
+				wantError: false,
```

Comment:
Do we need to specify the `wantError: false` here? Or can we remove it (the other tests do not have it). I think we can either add them to all, or remove them from all non-error test cases.

Reply:
Whoops, I was testing the struct declaration & forgot to get rid of that.

Reply:
If unspecified, it defaults to false. It is preferable to remove them for all non-error test cases :+1: